src/tqec/plaquette/rpng_description.py

```diff
@@ -0,0 +1,262 @@
+        if not r and r_str != '-':
+            raise ValueError('Unacceptable character for the R field.')
+        if not p and p_str != '-':
+            raise ValueError('Unacceptable character for the P field.')
+        if p and not a:
+            raise ValueError('Unacceptable ancilla basis in the extended format (i.e. rapng).')
+        if not n and n_str != '-':
+            raise ValueError('Unacceptable character for the N field.')
+        if not g and g_str != '-':
+            raise ValueError('Unacceptable character for the G field.')
+        return cls(r, p, n, g, a)
+
+    
+    def get_r_op(self) -> str | None:
+        """Get the reset operation or Hadamard"""
+        op = self.r
+        if op == None:
+            return None
+        elif op.value in BasisEnum._value2member_map_:
+            return f'R{op.value.upper()}'
+        else:
+            return f'{op.value.upper()}'
+
+    def get_g_op(self) -> str | None:
+        """Get the measurement operation or Hadamard"""
+        op = self.g
+        if op == None:
+            return None
+        elif op.value in BasisEnum._value2member_map_:
+            return f'M{op.value.upper()}'
+        else:
+            return f'{op.value.upper()}'
+
+
+@dataclass
+class RGN:
+    """Organize the prep and meas bases for the ancilla, together with the meas time
+    
+    The init time is assumed to be 0.
+
+    Attributes:
+        r   ancillaqubit reset basis  
+        g   ancilla qubit measure basis
+        n   time step of ancilla measurement
+    """
+    r: BasisEnum = BasisEnum.X
+    g: BasisEnum = BasisEnum.X
+    n: int = 6
+
+    @classmethod
+    def from_string(cls, rgn_string: str) -> 'RGN':
+        """Initialize the RGN object from a 3-character string"""
+        if len(rgn_string) != 3:
+            raise ValueError('The rgn string must be exactly 3-character long.')
+        r_str, g_str, n_str = tuple(rgn_string)
+
+        try:
+            r = BasisEnum(r_str)
+            g = BasisEnum(g_str)
+            n = int(n_str)
+            return cls(r, g, n)
+        except ValueError:
+            raise ValueError('Invalid rgn string.')
+
+
+@dataclass
+class RPNGDescription:
+    """Organize the description of a plaquette in RPNG format
+    
+    The corners of the square plaquette are listed following the order:
+    tl, tr, bl, br
+    If the ancilla RGN description is not specified, it is assumed 'xx6'
+
+    Attributes:
+        corners RPNG description of the four corners of the plaquette
+        ancilla RGN description of the ancilla
+    """
+    corners: tuple[RPNG, RPNG, RPNG, RPNG]
+    ancilla: RGN = RGN()
+
+    def __post_init__(self):
+        """Validation of the initialization arguments
+        
+        Constraints:
+        - the n values for the corners must be unique
+        - the n values for the corners must be in the interval ]0, ancilla.n[
+        - at least one betwween fields a and p in every corner is z
```

Comment:
These conditions should be removed for RPNG. Maybe for Extended RPNG, but I am unsure of its usefulness.

Reply:
I think it is important to validate the RPNG of the corners:
while each one can be correct in its own merit, further checks are possible when all corners are available.

Reply:
Removed